src/credentials.rs

```diff
@@ -31,55 +35,99 @@ pub struct Credentials {
     pub registry_tokens: HashMap<RegistryUri, String>,
 }
 
-impl Credentials {
-    fn location() -> eyre::Result<PathBuf> {
-        let home = home::home_dir().wrap_err("Failed to locate home directory")?;
+#[derive(Error, Display, Debug)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum LocateError {
+    /// the home folder could not be determined
+    MissingHome,
+}
+
+#[derive(Error, Display, Debug)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum ExistsError {
+    /// could not locate the credentials file
```

Comment:
Can we upercase the doc comments, and lowercase them manually in the wrapping error?

Reply:
The doc comments are now unified with the error messages, so unfortunately this stylistic conflict is something we have to live with. The alternative is to have duplicative comments and messages like before.